objects/src/block/note_tree.rs

```diff
@@ -55,20 +55,17 @@ impl BlockNoteTree {
         note_idx_in_batch: usize,
     ) -> Result<MerklePath, MerkleError> {
         let leaf_index =
-            LeafIndex::<NOTE_LEAF_DEPTH>::new(Self::note_index(batch_idx, note_idx_in_batch))?;
+            LeafIndex::<NOTE_LEAF_DEPTH>::new(Self::leaf_index(batch_idx, note_idx_in_batch))?;
 
         Ok(self.0.open(&leaf_index).path)
     }
 
     // HELPERS
     // --------------------------------------------------------------------------------------------
 
-    fn note_index(batch_idx: usize, note_idx_in_batch: usize) -> u64 {
-        (batch_idx * MAX_NOTES_PER_BATCH + note_idx_in_batch) as u64
-    }
-
-    fn leaf_index(batch_idx: usize, note_idx_in_batch: usize) -> u64 {
-        Self::note_index(batch_idx, note_idx_in_batch) * 2
+    /// Returns the leaf index for a Note. The metadata is located on the adjacent index.
+    pub fn leaf_index(batch_idx: usize, note_idx_in_batch: usize) -> u64 {
+        (batch_idx * MAX_NOTES_PER_BATCH * 2 + (note_idx_in_batch * 2)) as u64
```

Comment:
Is this correct? I'm not very familiar with how the note tree is structured but while this is equivalent to the previous code, doubling the `batch_idx` does not seem correct.

Reply:
I think the way you changed it is correct. Basically, `MAX_NOTES_PER_BATCH` is currently 4096 but since each note currently requires 2 leaves in the Merkle tree, the actual number of nodes in the tree is 8192. So, multiplying batch index by 8192 gives us the correct batch offset.

Similarly, doubling `note_idx_in_batch` is correct because note 0 would be at node 0, note 1 at node 2, note 2 at node 4 etc.

Reply:
Actually, looking at this more, I'm not sure we need this change. `note_index()` here should have given us the absolute index of note node in the block tree (i.e., the parent node of both the note leaf and the metadata leaf). This index would be for a path of length 20.

The `leaf_index()` would give us the index of note leaf at depth 21 - so, I think we should keep both.